driver/controller.go

```diff
@@ -191,6 +197,28 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	if snapshot != nil && volumeReq.SizeGigaBytes > int64(snapshot.SizeGigaBytes) && volumeReq.SizeGigaBytes > 1 {
+		log.Info("resizing volume because its requested size is larger than the size of the backing snapshot")
+		action, _, err := d.storageActions.Resize(ctx, vol.ID, int(volumeReq.SizeGigaBytes), volumeReq.Region)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "cannot resize volume %s: %s", vol.ID, err.Error())
+		}
+		log = log.WithFields(logrus.Fields{
+			"resized_from": int(snapshot.SizeGigaBytes),
+			"resized_to":   int(volumeReq.SizeGigaBytes),
+		})
+		if action != nil && action.Status != godo.ActionCompleted {
+			log = logWithAction(log, action)
+			log.Info("waiting until volume is resized")
+			waitActionCtx, waitActionCancel := context.WithDeadline(ctx, time.Now().Add(time.Second*15))
```

Comment:
IMHO, `context.WithTimeout` would allow for an easier specification of the actual timeout.

Regarding those 15 seconds: do we have a solid understanding by now of the time needed for a resize action to complete?